src/auth/auth.cpp

```diff
@@ -10,60 +10,87 @@
 
 #include <iosfwd>
 #include <optional>
+#include <ranges>
 #include <utility>
 
 #include <fmt/format.h>
 
 #include "auth/crypto.hpp"
 #include "auth/exceptions.hpp"
 #include "auth/rpc.hpp"
+#include "flags/auth.hpp"
 #include "license/license.hpp"
 #include "system/transaction.hpp"
 #include "utils/flag_validation.hpp"
 #include "utils/message.hpp"
 #include "utils/settings.hpp"
 #include "utils/string.hpp"
 
-DEFINE_VALIDATED_string(auth_module_executable, "", "Absolute path to the auth module executable that should be used.",
-                        {
-                          if (value.empty()) return true;
-                          // Check the file status, following symlinks.
-                          auto status = std::filesystem::status(value);
-                          if (!std::filesystem::is_regular_file(status)) {
-                            std::cerr << "The auth module path doesn't exist or isn't a file!" << std::endl;
-                            return false;
-                          }
-                          return true;
-                        });
-DEFINE_VALIDATED_int32(auth_module_timeout_ms, 10000,
-                       "Timeout (in milliseconds) used when waiting for a "
-                       "response from the auth module.",
-                       FLAG_IN_RANGE(100, 1800000));
+namespace memgraph {
+std::unordered_map<std::string, std::string> ModuleMappingsToMap(const std::string &module_mappings) {
+  std::unordered_map<std::string, std::string> module_per_scheme;
+  if (module_mappings.empty()) {
+    return module_per_scheme;
+  }
+
+  for (const auto &mapping : utils::Split(module_mappings, ";")) {
+    const auto module_and_scheme = utils::Split(mapping, ":");
+    const auto scheme_name = std::string{utils::Trim(module_and_scheme[0])};
+
+    const auto n_values_provided = module_and_scheme.size();
+    const auto use_default = n_values_provided == 1 && DEFAULT_SSO_MAPPINGS.contains(scheme_name);
+    if (n_values_provided != 2 && !use_default) {
+      throw auth::AuthException("Entries in the auth module mapping follow the \"auth_scheme: module_path\" syntax!");
+    }
```

Comment:
Can you add a comment with an example of the string we are expecting as the argument.
This is all very strict and confusing.

Reply:
I will, thanks!